From 1a190a8f4bd8c88dc9db0ea1206b4466b069a351 Mon Sep 17 00:00:00 2001
From: Carlos Magno <cmagnobarbosa@gmail.com>
Date: Sat, 23 Jan 2021 21:01:47 -0300
Subject: [PATCH 1/3] Update the default value `table_id` in the `FlowBase`
 class
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The default value of `table_id` 0xff (255) in flows is resulting in
an error in the consistency mechanism. This commit solves that,
changing this value to 0 instead of 255.

Co-authored-by: José Mauro <z.mauro@live.com>
---
 flow.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/flow.py b/flow.py
index 1292cf9..7433ecb 100644
--- a/flow.py
+++ b/flow.py
@@ -50,7 +50,7 @@ class FlowBase(ABC):  # pylint: disable=too-many-instance-attributes
     _flow_mod_class = None
     _match_class = None

-    def __init__(self, switch, table_id=0xff, match=None, priority=0x8000,
+    def __init__(self, switch, table_id=0x0, match=None, priority=0x8000,
                  idle_timeout=0, hard_timeout=0, cookie=0, actions=None,
                  stats=None):
         """Assign parameters to attributes.
@@ -58,7 +58,7 @@ def __init__(self, switch, table_id=0xff, match=None, priority=0x8000,
         Args:
             switch (kytos.core.switch.Switch): Switch ID is used to uniquely
                 identify a flow.
-            table_id (int): The index of a single table or 0xff for all tables.
+            table_id (int): The index of a single table.
             match (|match|): Match object.
             priority (int): Priority level of flow entry.
             idle_timeout (int): Idle time before discarding, in seconds.

From af15c1df879bcdbb33569735629178ce6019fff6 Mon Sep 17 00:00:00 2001
From: Carlos Magno <cmagnobarbosa@gmail.com>
Date: Mon, 8 Feb 2021 11:19:52 -0300
Subject: [PATCH 2/3] Add `table_id` field in `FlowMod`

---
 flow.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/flow.py b/flow.py
index 7433ecb..e332c7f 100644
--- a/flow.py
+++ b/flow.py
@@ -185,6 +185,7 @@ def _as_of_flow_mod(self, command):
         flow_mod.idle_timeout = self.idle_timeout
         flow_mod.hard_timeout = self.hard_timeout
         flow_mod.priority = self.priority
+        flow_mod.table_id = self.table_id
         return flow_mod

     @staticmethod

From ef85f178f4fbd46a55d72d6f2f46d60d45d00550 Mon Sep 17 00:00:00 2001
From: Carlos Magno <cmagnobarbosa@gmail.com>
Date: Thu, 4 Feb 2021 16:36:51 -0300
Subject: [PATCH 3/3] Apply suggestions from code review
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Co-authored-by: Humberto Diógenes <hdiogenes@gmail.com>
---
 flow.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/flow.py b/flow.py
index e332c7f..842a8d2 100644
--- a/flow.py
+++ b/flow.py
@@ -58,7 +58,8 @@ def __init__(self, switch, table_id=0x0, match=None, priority=0x8000,
         Args:
             switch (kytos.core.switch.Switch): Switch ID is used to uniquely
                 identify a flow.
-            table_id (int): The index of a single table.
+            table_id (int): The index of a single table or 0xff for all tables
+                            (0xff is valid only for the delete command)
             match (|match|): Match object.
             priority (int): Priority level of flow entry.
             idle_timeout (int): Idle time before discarding, in seconds.
